docs/source/app-dev/bindings-scala/index.rst

```diff
@@ -0,0 +1,144 @@
+verbosity level: **2** (INFO)::
+
+    $ java -jar codegen-main-100.12.17.jar scala ./quickstart-scala.dar=com.digitalasset.quickstart.iou.model \
+        --output-directory=./codegen-out --verbosity=2
+    ...
+    [INFO ] Scala Codegen result:
+    Number of generated templates: 3
+    Number of not generated templates: 0
+    Details:
+
+The output above tells that codegen produced Scala classes for 3 templates without errors (empty ``Details:`` line).
+
+Example code
+============
+
+This section refers to the IOU DAML example from the :doc:`Quickstart guide </getting-started/quickstart>` and
+`quickstart-scala sbt example <https://github.com/digital-asset/daml/tree/master/language-support/scala/examples/quickstart-scala>`_.
+
+Please keep in mind that all Scala examples compiled with ``-Xsource:2.13`` **scalac** option.
+
+Create a Contract and Send a Create Command
+-------------------------------------------
+
+To create a Scala class representing an ``Iou`` contract:
+::
+
+    import com.digitalasset.ledger.client.binding.{Primitive => P}
+    import com.digitalasset.quickstart.iou.model.{Iou => M}
+    ...
+    private val issuer = P.Party("Alice")
+    ...
+
+    val iou = M.Iou(
+      issuer = issuer,
+      owner = issuer,
+      currency = "USD",
+      amount = BigDecimal("1000.00"),
+      observers = List())
+
+To send a corresponding create command:
+::
+
+    val createCmd = iou.create
+    clientUtil.submitCommand(issuer, issuerWorkflowId, createCmd)
+
+For more details, please refer to the implementation of `com.digitalasset.quickstart.iou.ClientUtil#submitCommand <https://github.com/digital-asset/daml/blob/master/language-support/scala/examples/quickstart-scala/application/src/main/scala/com/digitalasset/quickstart/iou/ClientUtil.scala>`_.
+
+Receive a Transaction, Exercise a Choice and Send an Exercise Command
+---------------------------------------------------------------------
+
+To receive a transaction and decode a create event for ``IouTransfer`` contract:
+::
+
+    private val newOwner = P.Party("Bob")
+    ...
+    clientUtil.subscribe(newOwner, offset0, None) { tx =>
+      logger.info(s"$newOwner received transaction: $tx")
+      decodeCreated[M.IouTransfer](tx).foreach { contract: Contract[M.IouTransfer] =>
+        logger.info(s"$newOwner received contract: $contract")
+        ...
+      }
+    }
+
+To exercise ``IouTransfer_Accept`` choice on the received ``IourTransfer`` contract and send a corresponding exercise command:
+::
+
+    val exerciseCmd = contract.contractId.exerciseIouTransfer_Accept(actor = newOwner)
+    clientUtil.submitCommand(newOwner, newOwnerWorkflowId, exerciseCmd)
+
+Fore more details, please refer to the `com.digitalasset.quickstart.iou.IouMain#newOwnerAcceptsAllTransfers <https://github.com/digital-asset/daml/blob/master/language-support/scala/examples/quickstart-scala/application/src/main/scala/com/digitalasset/quickstart/iou/IouMain.scala>`_.
```

Comment:
Same as above - what will they learn from this? Be specific as to why they should follow the link, what it will help them with.